src/convertToObject.js

```diff
@@ -6,7 +6,16 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString
+    .split(';')
+    .map((value) => value.trim())
+    .filter((value) => value !== '')
+    .reduce((prev, string) => {
+      return {
+        ...prev,
+        [string.split(':')[0].trim()]: string.split(':')[1].trim(),
+      };
```

Comment:
[CODE KNOWLEDGE] - It's unnecessary to split the string twice inside the reduce function (lines 16 and 16). You can split it once, store the parts in variables, and then use them. This will improve performance since you won't be doing the same operation multiple times.